src/tribler/core/components/restapi/rest/rest_endpoint.py

```diff
@@ -14,16 +23,24 @@ class RESTEndpoint:
     def __init__(self, middlewares=()):
         self._logger = logging.getLogger(self.__class__.__name__)
         self.app = web.Application(middlewares=middlewares, client_max_size=2 * 1024 ** 2)
-        self.endpoints = {}
+        self.endpoints: Dict[str, RESTEndpoint] = {}
+        self.async_group = AsyncGroup()
         self.setup_routes()
 
     def setup_routes(self):
         pass
 
-    def add_endpoint(self, prefix, endpoint):
+    def add_endpoint(self, prefix: str, endpoint: RESTEndpoint | EventsEndpoint | IPV8RootEndpoint):
         self.endpoints[prefix] = endpoint
         self.app.add_subapp(prefix, endpoint.app)
 
+    async def shutdown(self):
+        for endpoint in self.endpoints.values():
+            if isinstance(endpoint, RESTEndpoint):  # IPV8RootEndpoint doesn't have a shutdown method
+                await endpoint.shutdown()
```

Comment:
Here we shut down endpoints one by one. I think shutting down all endpoints in parallel should be more efficient. Something like:
```python
shutdown_group = AsyncGroup()
for endpoint in self.endpoint.values():
    if isinstance(endpoint, RESTEndpoint):
        shutdown_group.add(endpoint.shutdown())
await shutdown_group.wait()
await self.async_group.cancel()
```

Reply:
Implemented